client/src/app/shared/progress-bar/progress-bar.service.ts

```diff
@@ -0,0 +1,28 @@
+import { Injectable } from '@angular/core';
+import { Subject, Observable } from 'rxjs';
+
+export enum ProgressBarEventType {
+  PROGRESS,
+  VISIBLE
+}
+
+export class ProgressBarEvent {
+  constructor(public type: ProgressBarEventType, public value: any) {}
+}
```

Comment:
How about we just go with an interface here?

```
interface ProgressBarEvent {
  eventType: ProgressBarEventType,
  value: any
}
```

And then we'd do

```
this.emitEvent({
  eventType: ProgressBarEventType.Visible,
  value: true
})
```

Maybe there's advantages to use a class as a type here that aren't apparent to me right now. But I fear the only thing it does is generate code that we don't actually need.

Happy to discuss this :)

Reply:
I agree it is useless to provide a class here. I changed it to a interface.